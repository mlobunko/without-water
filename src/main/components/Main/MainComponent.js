import React from 'react';
import { Switch } from 'react-router-dom';

import Header from 'app/Header';
import routes from '../../routes';
import RouteWithSubRoutes from '../RouteWithSubRoutes';

export const MainComponent = ({ authorized }) => (
  <div className="container-main">
    <Header />
    <Switch>
      {routes.map((route, i) => (
        <RouteWithSubRoutes authorized={authorized} key={i} {...route} />
      ))}
    </Switch>
  </div>
);
