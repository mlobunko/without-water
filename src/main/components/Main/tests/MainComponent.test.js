import React from 'react';
import { shallow } from 'enzyme';

import { MainComponent } from '../MainComponent';

describe('MainComponent', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<MainComponent authorized={true} />);
    expect(wrapper).toMatchSnapshot();
  });
});
