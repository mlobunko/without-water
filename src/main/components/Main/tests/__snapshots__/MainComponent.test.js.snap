// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainComponent renders correctly 1`] = `
<StyledMainComponent>
  <withRouter(Connect(HeaderContainer)) />
  <Switch>
    <Component
      authorized={true}
      exact={true}
      isRedirect={true}
      key="0"
      path="/"
      redirectTo="/without-water/news"
    />
    <Component
      authorized={true}
      exact={true}
      isRedirect={true}
      key="1"
      path="/without-water"
      redirectTo="/without-water/news"
    />
    <Component
      authorized={true}
      component={[Function]}
      key="2"
      path="/without-water/login"
    />
    <Component
      authorized={true}
      component={[Function]}
      key="3"
      path="/without-water/news"
    />
    <Component
      authorized={true}
      component={[Function]}
      isPrivate={true}
      key="4"
      path="/without-water/profile"
    />
    <Component
      authorized={true}
      component={[Function]}
      key="5"
    />
  </Switch>
</StyledMainComponent>
`;
